src/OpenTelemetry.Api/Internal/SelfDiagnostics.cs

```diff
@@ -0,0 +1,72 @@
+﻿// <copyright file="SelfDiagnostics.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Internal
+{
+    /// <summary>
+    /// Self diagnostics class captures the EventSource events sent by OpenTelemetry
+    /// modules and writes them to local file for troubleshooting.
+    /// </summary>
+    public class SelfDiagnostics : IDisposable
```

Comment:
Given this is intended to be a public API, I guess it needs to go to `OpenTelemetry` namespace rather than `OpenTelemetry.Internal`.

Reply:
After second thoughts, I think it's better to limit the `SelfDiagnostics` class to only capture (with EventListener) and log events from existing EventSource implementations. So it can be viewed as a feature of OpenTelemetry SDK, rather than a public API in OpenTelemetry API. And the class would fall in the `src/OpenTelemetry/Internal` folder in namespace `OpenTelemetry.Internal` with `internal` access level.

As for providing public logging API for all to use, it might take longer to finalize the new API. I think I should just start a proposal for now according to the "Changes in public API reviewed" suggestion in the PR template. This way I can start working on capturing existing events before the logging APIs are finalized in design discussions.